### PR TITLE
SDKS-2143 Add expiration time to JWS for Device Binding and Device Signing Verifier

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket

[Please link jira ticket here](https://bugster.forgerock.org/jira/browse/SDKS-2143)

# Description

Add expiration time to JWS for Device Binding and Device Signing Verifier

# Definition of Done Checklist:

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Ensure backward compatibility.
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] Functional spec is written/updated.
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

### CODE REVIEW

**Review by spetrov — DISMISSED** [review]
LGTM 👍🏻

File: forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt [hunk 1]
```diff
@@ -69,24 +73,32 @@ interface DeviceAuthenticator {
...
         signedJWT.sign(RSASSASigner(keyStoreKey))
         return signedJWT.serialize()
     }
+
     /**
      * check biometric is supported
      */
     fun isSupported(): Boolean
 
+    fun getDefaultExpirationTime(): Date {
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt:96]: lets talk about this

File: forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt [hunk 2]
```diff
@@ -96,18 +108,14 @@ interface DeviceAuthenticator {
...
+data class KeyPair(val publicKey: RSAPublicKey, val privateKey: PrivateKey, var keyAlias: String)
 
 /**
  * Settings  for all the biometric authentication is configured
  */
 internal class BiometricOnly(private val biometricInterface: BiometricHandler,
-                    private val keyAware: KeyAware,
-                    private val isApi30OrAbove: Boolean): DeviceAuthenticator {
+                             private val keyAware: KeyAware,
+                             private val isApi30OrAbove: Boolean) : DeviceAuthenticator {
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt:118]: private val isApi30OrAbove: Boolean) : DeviceAuthenticator - Remove Space
 
like this

private val isApi30OrAbove: Boolean):  DeviceAuthenticator - Remove Space

File: forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt [hunk 3]
```diff
@@ -146,8 +155,9 @@ internal class BiometricOnly(private val biometricInterface: BiometricHandler,
  * Settings for all the biometric authentication and device credential is configured
  */
 internal class BiometricAndDeviceCredential(private val biometricInterface: BiometricHandler,
-                                   private val keyAware: KeyAware,
-                                   private val isApi30OrAbove: Boolean): DeviceAuthenticator {
+                                            private val keyAware: KeyAware,
+                                            private val isApi30OrAbove: Boolean) :
+    DeviceAuthenticator {
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt:160]: private val isApi30OrAbove: Boolean) : DeviceAuthenticator - Remove Space
 
like this

private val isApi30OrAbove: Boolean):  DeviceAuthenticator - Remove Space

File: forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt [hunk 4]
```diff
@@ -185,7 +197,7 @@ internal class BiometricAndDeviceCredential(private val biometricInterface: Biom
 /**
  * Settings for all the none authentication is configured
  */
-internal class None(private val keyAware: KeyAware): DeviceAuthenticator {
+internal class None(private val keyAware: KeyAware) : DeviceAuthenticator {
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt:200]: internal class None(private val keyAware: KeyAware) : DeviceAuthenticator - Remove Space
 
like this

internal class None(private val keyAware: KeyAware): DeviceAuthenticator

**Review by jeyanthanperiyasamy — CHANGES_REQUESTED** [review]
minor feedback

File: forgerock-auth/src/main/java/org/forgerock/android/auth/interceptor/IdentityGatewayAdviceInterceptor.java [hunk 5]
```diff
@@ -16,6 +16,8 @@
 import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
+import java.net.HttpURLConnection;
```
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/interceptor/IdentityGatewayAdviceInterceptor.java:19]: Update copyright

Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/interceptor/IdentityGatewayAdviceInterceptor.java]: This file should not be here, rollback changes.

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

**Review by jeyanthanperiyasamy — APPROVED** [review]

**Pull request merged**